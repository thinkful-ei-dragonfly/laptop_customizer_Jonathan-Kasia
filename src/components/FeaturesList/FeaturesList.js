import React from 'react';
import Feature from './Feature/Feature'

function FeaturesList(props){
  const features = Object.keys(props.features)
          .map(key => {
            const options = props.features[key].map((item, index) => {
              const selectedClass = item.name === props.selected[key].name ? 'feature__selected' : '';
              const featureClass = 'feature__option ' + selectedClass;
              return <Feature key={index} item={item} featureClass={featureClass} updateFeature={props.updateFeature} />
            });

            return <div className="feature" key={key}>
              <div className="feature__name">{key}</div>
              <ul className="feature__list">
                { options }
              </ul>
            </div>
          });      
  
  
  return (
    <section className="main__form">
            <h3>TECH SPECS AND CUSTOMIZATIONS</h3>
            {features}
          </section>
         
  )
}

export default FeaturesList;