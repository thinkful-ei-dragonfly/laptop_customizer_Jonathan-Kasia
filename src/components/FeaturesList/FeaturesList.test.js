import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FeaturesList from './FeaturesList'

describe('Features List component', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<FeaturesList />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders the UI as expected', () =>{
    const tree = renderer
    .create(<FeaturesList />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
})