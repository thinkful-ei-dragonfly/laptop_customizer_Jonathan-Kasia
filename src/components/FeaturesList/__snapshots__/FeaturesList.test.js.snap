// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features List component Renders the UI as expected 1`] = `
<section
  className="main__form"
>
  <h3>
    TECH SPECS AND CUSTOMIZATIONS
  </h3>
</section>
`;
